ngraph/core/reference/include/ngraph/runtime/reference/roi_align.hpp

```diff
@@ -0,0 +1,243 @@
+                                    auto s3 = feature_maps[feature_maps_transform.index(
+                                        {static_cast<uint64_t>(batch_indices[roi_index]),
+                                         channel_index,
+                                         pooling_points[sample_index + 2].first,
+                                         pooling_points[sample_index + 2].second})];
+                                    auto s4 = feature_maps[feature_maps_transform.index(
+                                        {static_cast<uint64_t>(batch_indices[roi_index]),
+                                         channel_index,
+                                         pooling_points[sample_index + 3].first,
+                                         pooling_points[sample_index + 3].second})];
+
+                                    switch (pooling_mode)
+                                    {
+                                    case ROIPoolingMode::MAX:
+                                    {
+                                        T sample_value =
+                                            std::max({pooling_weights[sample_index] * s1,
+                                                      pooling_weights[sample_index + 1] * s2,
+                                                      pooling_weights[sample_index + 2] * s3,
+                                                      pooling_weights[sample_index + 3] * s4});
+
+                                        pooled_value = sample_value > pooled_value ? sample_value
+                                                                                   : pooled_value;
+                                        break;
+                                    }
+                                    case ROIPoolingMode::AVG:
+                                    default:
+                                    {
+                                        T sample_value = pooling_weights[sample_index] * s1 +
+                                                         pooling_weights[sample_index + 1] * s2 +
+                                                         pooling_weights[sample_index + 2] * s3 +
+                                                         pooling_weights[sample_index + 3] * s4;
+                                        pooled_value += sample_value / (num_samples_in_bin);
+                                    }
+                                    }
+                                    sample_index += 4;
+                                }
+                                tmp_out.push_back(pooled_value);
+                            }
+                        }
+                        // save the calculations for all bins across this channel
+                        std::copy(tmp_out.begin(),
+                                  tmp_out.end(),
+                                  &out[out_transform.index({static_cast<uint64_t>(roi_index),
```

Comment:
Can you move the index method call somewhere above and assign it to a variable called `offset`? Then you can use this variable in the call to `std::copy` like `out + offset`.

Reply:
Done